js/saveFormData.js

```diff
@@ -0,0 +1,15 @@
+const contactUsForm = document.forms[0];
+const formDetails = {};
+if (window.localStorage.getItem('data')) {
+  const parsedData = JSON.parse(window.localStorage.getItem('data'));
+  contactUsForm.name.value = parsedData.name;
+  contactUsForm.email.value = parsedData.email;
+  contactUsForm.message.value = parsedData.message;
+}
+function collectFormData() {
+  formDetails.name = contactUsForm.name.value;
+  formDetails.email = contactUsForm.email.value;
+  formDetails.message = contactUsForm.message.value;
+  window.localStorage.setItem('data', JSON.stringify(formDetails));
+}
+contactUsForm.addEventListener('submit', collectFormData);
```

Comment:
- [ ] Good job persisting the form datas in the local storage 😄. Kindly ensure that after filling out the form and not submitting it, the entered values should be saved in the browser.

I can see that you are only storing to local storage when the form is submitted. The values should be saved in the local storage even without submitting the form.

Reply:
@vickymarz thanks a lot for mentioning it I did not know it is required to save every change this can be achieved by replacing 'submit' with 'change' in the form event listener.

Reply:
👍